Lib/distutils/command/install.py

```diff
@@ -281,12 +281,10 @@ def finalize_options(self):
         # about needing recursive variable expansion (shudder).
 
         py_version = sys.version.split()[0]
-        (prefix, exec_prefix) = get_config_vars('prefix', 'exec_prefix')
-        try:
-            abiflags = sys.abiflags
-        except AttributeError:
-            # sys.abiflags may not be defined on all platforms.
-            abiflags = ''
+        prefix, exec_prefix, abiflags = get_config_vars('prefix',
+                                                'exec_prefix', 'ABIFLAGS')
+        # sys.abiflags may not be defined on all platforms.
+        abiflags = '' if abiflags is None else abiflags
```

Comment:
Why not using sys.abiflags when cross-compilation is not used?

Above, sys.prefix and sys.exec_prefix are used for prefix and exec_prefix when cross compilation is not used. Maybe do the same here?

Reply:
The current code in the master branch uses `get_config_vars()` to get `prefix` and `exec_prefix` whether it is for cross-compilation or not. So for consistency the PR does the same for abiflags. When not cross-compiling `sys.abiflags` and `abiflags` from the sysconfigdata module are the same since their value come from the Makefile so it does not make any difference.

However there is a bug here (in the current Python code) as `sys.prefix` and `sys.exec_prefix` set by `getpath.c` may be different in some rare cases from their values in the sysconfigdata module, for example when an installed python is manually copied somewhere else.

I will enter a new issue.

Reply:
This is the new issue https://bugs.python.org/issue38949

Reply:
Thanks for opening the other issue.